### PR TITLE
Feature/swift4x API modernisation

### PR DESCRIPTION
**Pull request by PaulEhrhardt**
# Swift4.2 support and API modernisation

--- 

* [x] slicker public API
* [x] full Obj-C support
* [x] configuration of policies via settings object
* [x] full code coverage by tests
* [x] updated readme
* [x] updated examples

---

### CODE REVIEW

File: GCXTrustPolicy/TrustPolicy.swift [hunk 1]
```diff
@@ -18,50 +18,153 @@
...
 
-/// Protocol definition for validating a policy against a remote trust
+/// Type alias for a closre that provides custom validation
+public typealias CustomValidationClosure = (SecTrust?) -> (Bool)
+
+@objc(GCXValidationType)
+/// Trust policy validation types.
+///
+/// - disabled: Performs no validation at all.
+///				It is advised to be careful with disabling validation
```
Comment (virtualrapha) [inline GCXTrustPolicy/TrustPolicy.swift:28]: Komische Einrückung hier im kompletten Kommentarblock 😅

**Review by virtualrapha — APPROVED** [review]
Schick geworden! LGTM!

**Review by snod — APPROVED** [review]
LGTN

**Comment by virtualrapha** [comment]
@PaulEhrhardt Können wir hier mergen?

**Pull request merged**